tests/jsunit/generator_test.js

```diff
@@ -19,10 +19,85 @@
  */
 'use strict';
 
+goog.require('Blockly.Dart');
+goog.require('Blockly.JavaScript');
+goog.require('Blockly.Lua');
+goog.require('Blockly.PHP');
+goog.require('Blockly.Python');
+
+var workspace;
+
+function defineGeneratorTestBlocks() {
+  Blockly.defineBlocksWithJsonArray([{
+    "type": "stack_block",
+    "message0": "",
+    "previousStatement": null,
+    "nextStatement": null
+  },
+  {
+    "type": "row_block",
+    "message0": "%1",
+    "args0": [
+      {
+        "type": "input_value",
+        "name": "INPUT"
+      }
+    ],
+    "output": null,
+    "nextStatement": null
+  }]);
+}
+
+function undefineTestBlocks() {
+  delete Blockly.Blocks['stack_block'];
+  delete Blockly.Blocks['row_block'];
+}
+
+function generatorTest_setUp() {
+  defineGeneratorTestBlocks();
+  workspace = new Blockly.Workspace();
+}
+
+function generatorTest_tearDown() {
+  undefineTestBlocks();
+  workspace.dispose();
+}
+
+function blockToCodeTest(generator, opt_thisOnly) {
+  var row_block = workspace.newBlock("row_block");
+  var stack_block = workspace.newBlock("stack_block");
+
+  generator.row_block = (block) => {return "row_block"};
+  generator.stack_block = (block) => {return "stack_block"};
```

Comment:
To my knowledge, this would represent the only use of ES6 in Blockly.  Since this is a test, it probably won't break the compiler's settings.  None the less, it's probably safer to just use ES5 here.